tests/project/test_brownie_config.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python3
+import os
+
+import pytest
+import yaml
+
+from brownie._config import _get_data_folder, _load_config
+from brownie.network.rpc import _validate_cmd_settings
+
+
+@pytest.fixture
+def settings_proj(testproject):
+    """Creates a config file in the testproject root folder and loads it manually."""
+
+    # Save the following config as "brownie-config.yaml" in the testproject root
+    test_brownie_config = """
+    networks:
+        default: development
+        development:
+            gas_limit: 6543210
+            gas_price: 1000
+            reverting_tx_gas_limit: 8765432
+            default_contract_owner: false
+            cmd_settings:
+                gas_limit: 7654321
+                block_time: 5
+                default_balance: 15 milliether
+                time: 2019-04-05T14:30:11Z
+                accounts: 15
+                evm_version: byzantium
+                mnemonic: brownie2
+    """
+    with testproject._path.joinpath("brownie-config.yaml").open("w") as fp:
+        yaml.dump(yaml.load(test_brownie_config), fp)
+
+    # Load the networks.development config from the created file and yield it
+    with testproject._path.joinpath("brownie-config.yaml").open() as fp:
+        conf = yaml.safe_load(fp)["networks"]["development"]
+        yield conf
+
+    os.remove(testproject._path.joinpath("brownie-config.yaml"))
```

Comment:
`testproject` creates a fresh project in a new folder for every test that uses it, so there's no need for cleaning up the filesystem like this.